### PR TITLE
chore(deps): update angular and related packages to v14

### PR DESCRIPTION
**Pull request by dawidgil**
pr for task: https://energyweb.atlassian.net/browse/SWTCH-2196

### CODE REVIEW

**Comment by dawidgil** [comment]
@jrhender there are few `as any` because Angular v14 supports typed form. However we have some forms created on the fly, which creates issues with typing. If you prefer i can add unit tests + refactor/rewrite those forms to have proper typing. Do you want from me to rewrite in this PR or should I leave it as it is?

File: angular.json [hunk 1]
```diff
@@ -226,32 +226,12 @@
...
-                  "with": "src/environments/environment.dev.ts"
-                }
-              ],
-              "optimization": true,
-              "outputHashing": "all",
-              "sourceMap": true,
-              "namedChunks": false,
-              "extractLicenses": true,
-              "vendorChunk": false,
-              "buildOptimizer": true,
```
Comment (jrhender) [inline angular.json]: Is this removal intentional? Is it related to angular 14 upgrade?

File: src/app/routes/applications/new-role/new-role.component.spec.ts [hunk 2]
```diff
@@ -60,22 +60,6 @@ describe('NewRoleComponent', () => {
...
-          [{ type: PreconditionType.Role, conditions: ['a', 'b', 'c', 'd'] }],
-        ],
-      }),
-    });
-    component.removePreconditionRole(index);
-    expect(
-      component.roleForm.get('data').get('enrolmentPreconditions').value[0]
-        .conditions
-    ).toEqual(['a', 'c', 'd']);
-  });
```
Comment (jrhender) [inline src/app/routes/applications/new-role/new-role.component.spec.ts:77]: Do these tests no longer make sense in Angular 14?

File: src/app/routes/registration/request-claim/request-claim.component.ts [hunk 3]
```diff
@@ -276,9 +281,10 @@ export class RequestClaimComponent implements OnInit, SubjectElements {
         registrationTypes: enrolForm.registrationTypes,
       } as any);
 
+      // TODO: remove any, add proper typing to the form
       this.displayAlert(
         'Request to enrol as ' +
-          this.roleTypeForm.value.roleType.name.toUpperCase() +
+          (this.roleTypeForm.value.roleType as any).name.toUpperCase() +
```
Comment (jrhender) [inline src/app/routes/registration/request-claim/request-claim.component.ts]: Would the way to fix this be to have `roleType` in the `IRoleTypeForm` interface be `{ name: string }`?

File: src/app/shared/expiration/date-picker/expiration-date.component.ts [hunk 4]
```diff
@@ -91,7 +91,7 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
     }
   }
 
-  private calcSeconds(value: string): number {
+  private calcSeconds(value: Date): number {
     const d = new Date(value);
```
Comment (jrhender) [inline src/app/shared/expiration/date-picker/expiration-date.component.ts]: Can this line be removed if the param type is `Date`?
```suggestion
  private calcSeconds(value: Date): number {
```
By the way, is this change related to the angular 14 upgrade?

**Review by jrhender — COMMENTED** [review]
> there are few as any because Angular v14 supports typed form. However we have some forms created on the fly, which creates issues with typing. If you prefer i can add unit tests + refactor/rewrite those forms to have proper typing. Do you want from me to rewrite in this PR or should I leave it as it is?

In my opinion, we can leave it as is (with the `any`s) and add typings in another PR. My rationale is that we didn't have typings before so upgrading without typings doesn't leave us in a worse state and lets us have smaller PRs. How does that sound?

File: angular.json (hunk 1 above)
Comment (dawidgil) [reply angular.json]: No, it is not related to angular 14. I should revert this change, as I fixed this in source maps for sentry. I wrongly updated this file previously, so running `ng serve` was starting local version with `prod` flag.

File: src/app/routes/applications/new-role/new-role.component.spec.ts (hunk 2 above)
Comment (dawidgil) [reply src/app/routes/applications/new-role/new-role.component.spec.ts:77]: Angular was complaining on those forms in tests. Also for me, those tests doesn't make sense, because they are not testing component but a specific function. Imo we shouldn't create forms like this in spec files.

File: src/app/shared/expiration/date-picker/expiration-date.component.ts (hunk 4 above)
Comment (dawidgil) [reply src/app/shared/expiration/date-picker/expiration-date.component.ts]: yes, it was complaining on `string` here. 
Checked: it is NOT safe to remove that Date creation. User can pass string to the input (he doesn't need to use callendar). If he pass string, then we do not have Date object, we have string. I will adjust typing that it might be `Date | string`

File: src/app/routes/registration/request-claim/request-claim.component.ts (hunk 3 above)
Comment (dawidgil) [reply src/app/routes/registration/request-claim/request-claim.component.ts]: Ok, found out that this roleType is `IRole` from icl. I won't add that to `IRoleTypeForm` as we need only `name` property. So i will just set `name` from `IRole` and there is no need to have `name` in form.

File: src/app/routes/applications/new-role/new-role.component.spec.ts (hunk 2 above)
Comment (jrhender) [reply src/app/routes/applications/new-role/new-role.component.spec.ts:77]: Ok, sounds good

**Review by jrhender — APPROVED** [review]
Thanks, looks good
